Lib/test/_test_multiprocessing.py

```diff
@@ -3610,6 +3616,246 @@ def test_copy(self):
+                        flags=shared_memory.O_CREX,
+                        size=5000
+                    )
+
+                    try:
+                        self.assertGreaterEqual(sms_uno.size, 5000)
+
+                        sms_duo = shared_memory.SharedMemory('test01_dblunlink')
+                        sms_duo.unlink()  # First shm_unlink() call.
+                        sms_duo.close()
+                        sms_uno.close()
+
+                    finally:
+                        sms_uno.unlink()  # A second shm_unlink() call is bad.
+
+            # Enforcement of `mode` and `read_only` is OS platform dependent
+            # and as such will not be tested here.
+
+            with self.assertRaises(shared_memory.ExistentialError):
+                # Attempting to create a new shared memory segment with a
+                # name that is already in use triggers an exception.
+                there_can_only_be_one_sms = shared_memory.SharedMemory(
+                    'test01_tsmb',
+                    flags=shared_memory.O_CREX,
+                    size=512
+                )
+
+            # Requesting creation of a shared memory segment with the option
+            # to attach to an existing segment, if that name is currently in
+            # use, should not trigger an exception.
+            # Note:  Using a smaller size could possibly cause truncation of
+            # the existing segment but is OS platform dependent.  In the
+            # case of MacOS/darwin, requesting a smaller size is disallowed.
+            ok_if_exists_sms = shared_memory.SharedMemory(
+                'test01_tsmb',
+                flags=shared_memory.O_CREAT,
+                size=sms.size if sys.platform != 'darwin' else 0
+            )
+            self.assertEqual(ok_if_exists_sms.size, sms.size)
+            ok_if_exists_sms.close()
+
+            # Attempting to attach to an existing shared memory segment when
+            # no segment exists with the supplied name triggers an exception.
+            with self.assertRaises(shared_memory.ExistentialError):
+                nonexisting_sms = shared_memory.SharedMemory('test01_notthere')
+                nonexisting_sms.unlink()  # Error should occur on prior line.
+
+            sms.close()
+
+        finally:
+            # Prevent test failures from leading to a dangling segment.
+            sms.unlink()
+
+    def test_shared_memory_across_processes(self):
+        sms = shared_memory.SharedMemory(
+            'test02_tsmap',
+            flags=shared_memory.O_CREX,
+            size=512
+        )
+
+        try:
+            p = self.Process(
+                target=self._attach_existing_shmem_then_write,
+                args=(sms.name, b'howdy')
+            )
+            p.daemon = True
+            p.start()
+            p.join()
+            self.assertEqual(bytes(sms.buf[:5]), b'howdy')
+
+            sms.close()
+
+        finally:
+            sms.unlink()
```

Comment:
You could use `self.addCleanup(sms.unlink)` right after creating `sms`, spare this long try/finally block and save one indentation.

Reply:
Good thought -- same as prior comment.